.stylelintrc.json

```diff
@@ -52,6 +52,10 @@
       "ignoreKeywords": ["inherit", "transparent"]
     }],
 
-    "declaration-empty-line-before": null
+    "declaration-empty-line-before": null,
+
+    "scss/dollar-variable-empty-line-before": ["always", {
+      "except": ["first-nested", "after-comment", "after-dollar-variable"]
```

Comment:
I think this needs to be split up into `"except": ["first-nested"]` because we never want an empty-line on the first nested element)and `"ignore": ["after-comment", "after-dollar-variable"]` so both empty-line and no empty-line are allowed after a comment or a variable